Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -151,34 +154,48 @@
         }
     }
 
-    NSData* requestBody = nsRequest.HTTPBody;
-    if (nsRequest.HTTPBodyStream) {
+    ComPtr<IHttpContent> httpContent;
+    if (NSData* requestBody = nsRequest.HTTPBody) {
+        ComPtr<IBuffer> bodyBuffer;
+        THROW_IF_FAILED(BufferFromRawData(bodyBuffer.GetAddressOf(),
+                                          reinterpret_cast<unsigned char*>(const_cast<void*>([requestBody bytes])),
+                                          [requestBody length]));
+
+        ComPtr<IHttpBufferContentFactory> httpBufferContentFactory;
+        THROW_IF_FAILED(GetActivationFactory(Wrappers::HStringReference(RuntimeClass_Windows_Web_Http_HttpBufferContent).Get(),
+                                             &httpBufferContentFactory));
+        THROW_IF_FAILED(httpBufferContentFactory->CreateFromBuffer(bodyBuffer.Get(), &httpContent));
+        THROW_IF_FAILED(httpRequestMessage->put_Content(httpContent.Get()));
+
+    } else if (NSInputStream* bodyStream = nsRequest.HTTPBodyStream) {
         // VSO 6693048: Properly support streaming bodies instead of flattening the data up front.
```

Comment:
Burned a few days trying to fix this as well, but the threading turns out to be super difficult to get right without C++/CX-y or C#-y language async support (Tried to implement own IASyncOperationWithProgress, tripped over threading requirements). Decided to cut losses and just go with this.